internal/engine/filter/convert.go

```diff
@@ -228,13 +229,18 @@ func joinQueries(f1, f2 string) string {
 
 func joinStrings(escape bool, a ...string) string {
 	if escape {
-		return joinEscaped(a...)
+		b, _ := joinEscaped(a...)
```

Comment:
I think it's easier to add a check here for whether any of the strings to escape are already escaped. This way you don't have to plumb the error propagation to too many functions.

```
if escape {
    // If the strings are already quoted, we would double quote here.
    // This is not recognized correctly by Gmail and we should fail.
    for _, a := range a {
        if strings.Contains(a, `"`) {
            return "", fmt.Errorf("invalid quote in %q.", a)
        }
    }
    return joinEscaped(a...), nil
}
```